forc-test/src/lib.rs

```diff
@@ -641,17 +672,23 @@ fn deployment_transaction(
     (contract_id, tx)
 }
 
-/// Build the given package and run its tests, returning the results.
-fn run_tests(built: BuiltTests, test_runners: &rayon::ThreadPool) -> anyhow::Result<Tested> {
+/// Build the given package and run its tests after applying the filter provided.
+///
+/// Returns the result of test execution.
+fn run_tests(
+    built: BuiltTests,
+    test_runners: &rayon::ThreadPool,
+    test_filter: Option<String>,
+) -> anyhow::Result<Tested> {
     match built {
         BuiltTests::Package(pkg) => {
-            let tested_pkg = pkg.run_tests(test_runners)?;
+            let tested_pkg = pkg.run_tests(test_runners, test_filter.as_deref())?;
```

Comment:
Nice point! It should not (and also does not as we have our unit test e2es still passing without a filter), you can see that from [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ff0cf2537b83f7d43340d6a02218f34b) playground. 

It is because, in core, `as_deref` is implemented for `Option` like the following:

```rust
pub const fn as_deref(&self) -> Option<&T::Target>
where
    T: ~const Deref,
{
    match self.as_ref() {
        Some(t) => Some(t.deref()),
        None => None,
    }
}
```

[reference](https://doc.rust-lang.org/src/core/option.rs.html#1267-1277)